### PR TITLE
Refactor vs package restore hide vsinterfaces

### PR DESCRIPTION
**Pull request by ocallesp**
This is part of https://github.com/dotnet/project-system/issues/8574

Interfaces coming from the NuGet Client (e.g. [IVsProjectRestoreInfo2](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.VisualStudio/SolutionRestoreManager/IVsProjectRestoreInfo2.cs)) are using the attribute `ComImport`, but this attribute is irrelevant since these interfaces are not pointing to any COM object anymore. This makes the interfaces independent of Vs and the implementations can be moved to `ProjectSystem.Managed` project

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/8640)

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/VSResources.resx [hunk 1]
```diff
@@ -171,15 +171,6 @@ In order to debug this project, add an executable project to this solution which
   <data name="ImportsFromTargetCannotBeDeleted" xml:space="preserve">
     <value>{0} - Import comes fom target file. This import cannot be removed.</value>
   </data>
-  <data name="Restore_PropertyWithInconsistentValues" xml:space="preserve">
```
[the remark this answers is not in the document]
Comment (ocallesp) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/VSResources.resx:174]: These string resources were moved from `src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/VSResources.resx`  to `src/Microsoft.VisualStudio.ProjectSystem.Managed/Resources.resx` because the code that is using this string resource was moved to ProjectSystem.Managed project.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Microsoft.VisualStudio.ProjectSystem.Managed.VS.csproj [hunk 2]
```diff
@@ -76,6 +76,9 @@
 
   <ItemGroup>
     <None Include="**\*.xlf" />
+    <Compile Remove="ProjectSystem\VS\PackageRestore\Snapshots\**" />
+    <EmbeddedResource Remove="ProjectSystem\VS\PackageRestore\Snapshots\**" />
+    <None Remove="ProjectSystem\VS\PackageRestore\Snapshots\**" />
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Microsoft.VisualStudio.ProjectSystem.Managed.VS.csproj:81]: Looks like these paths don't exist any more. Are these lines needed?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/Resources.resx [hunk 3]
```diff
@@ -127,7 +127,7 @@
   <data name="JsonErrorsNeedToBeCorrected_WithErrorMessage_2" xml:space="preserve">
     <value>Errors in '{0}' need to be corrected before applying changes. Error message:
 {1}</value>
-   <comment>{0} is the path to the malformed .json file. {1} is an error message containing more details about the problems in the file.</comment>
+    <comment>{0} is the path to the malformed .json file. {1} is an error message containing more details about the problems in the file.</comment>
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/Resources.resx:130]: This seems to be an unrelated change.

**Comment by drewnoakes** [comment]
@nkolev92 the premise of this PR is that type `IVsProjectRestoreInfo2` (and therefore `IVsTargetFrameworks2` and `IVsReferenceItems`) are not tied to VS, despite their names. Does that sound right to you?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/Resources.resx (hunk 3 above)
Comment (ocallesp) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/Resources.resx]: removed

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Microsoft.VisualStudio.ProjectSystem.Managed.VS.csproj (hunk 2 above)
Comment (ocallesp) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Microsoft.VisualStudio.ProjectSystem.Managed.VS.csproj]: removed

**Comment by nkolev92** [comment]
> @nkolev92 the premise of this PR is that type `IVsProjectRestoreInfo2` (and therefore `IVsTargetFrameworks2` and `IVsReferenceItems`) are not tied to VS, despite their names. Does that sound right to you?

Yep. It's just convention naming that we followed. The API no dependencies, it's only NuGet.

**Review by drewnoakes — APPROVED** [review]
